sig-scalability/processes/formal-scalability-processes.md

```diff
@@ -0,0 +1,136 @@
+
+## Processes in Detail
+
+### Implementation / Pre-submit phase
+
+A good design does not always mean a good implementation. Besides, code changes are made that do not necessarily correspond to some feature. So we need checks in place for new code being checked-in to the repo, to ensure system scalability is not harmed. For this purpose, we would add two kinds of pre-submit scalability jobs running performance tests:
+
+#### Optional PR jobs
+
+These are scalability jobs that do not automatically run against PRs. They need 
+to be manually triggered using a bot command and are expected to be triggered by PR authors or reviewers (or delegated to sig-scalability if needed) if they believe the changes can affect scalability. These would further be of two kinds:
+
+- Medium-scale jobs\
+  ([relevant feature request](https://github.com/kubernetes/test-infra/issues/5677))\
+  Jobs that do not consume too many resources (light on quota-usage), not that high on cost and time and low-risk wrt outages, etc. These can be triggered by ~anybody. The following would be available here (can change over time):
+
+  - Kubemark-500 (consumes ~80 vCPUs)
+  - GCE-100 (consumes ~100 vCPUs)
+
+- Large-scale jobs\
+  ([relevant feature request](https://github.com/kubernetes/test-infra/issues/5051))\
+  Jobs that are much heavier wrt resources, time, cost and are high-risk. Their trigger would be access-restricted to a group of few googlers for those reasons. To begin with, the group would consist of key googlers from sig-scalability. Also these jobs need quota-checking before trigger and are to be used sparingly. The following would be available here (can change over time):
+
+  - Kubemark-2000 (consumes ~270 vCPUs)
+  - Kubemark-5000 (consumes ~700 vCPUs)
+  - GCE-500 (consumes ~520 vCPUs)
+  - GCE-2000 (consumes ~2030 vCPUs)
+
+#### Mandatory pre-merge jobs
+([relevant feature request](https://github.com/kubernetes/test-infra/issues/4445))
+
+These are scalability jobs that would run against the PR at the head of the submit-queue right before merging. Note that they wouldn’t be running by default against each opened PR due to resource, cost and other constraints. Further, these should be fast and healthy enough in order to not hurt our merge rate. If the failure rate of their corresponding post-submit jobs goes >2% (measured over last 1 week), it would lead to a P0 issue needing immediate attention from sig-scalability. If needed, this metric may be modified later to subtract false positives (i.e failures belonging to a regression that has been fixed later). Next, these jobs should also be available and open for the author of a PR to be able to kick them - as they might want to verify a fix in case the pre-merge job failed on their PR. The following would be available here (can change over time):
```

Comment:
If they do not run on every PR, when do they run?

Reply:
The idea is to try to run those only during pre-merge round (i.e not in the initial round of pre-submits which happens for ~every single PR opened and might be a waste). There is some discussion going around how this might cause inconsistencies in our current SQ frameowrk under https://github.com/kubernetes/test-infra/issues/4445. Hopefully we reach some consensus there.